src/data/markdown/docs/01 guides/05 Testing Guides/04 Large-scale k6 testing.md

```diff
@@ -0,0 +1,484 @@
+
+Below we will explore what hardware is needed for generating different levels of load.
+
+
+<div class="doc-blockquote">
+
+> ### Note about native distributed-execution in k6
+>
+> The long-term goal for k6 is to support distributed execution natively. We are currently laying groundwork for this feature in [PR #1007](https://github.com/loadimpact/k6/pull/1007).
+> You can follow this effort on guthub [in issue 140](https://github.com/loadimpact/k6/issues/140). Again, this is useful only if you need to generate load larger than 300k RPS.
+>
+> [k6 cloud](/cloud) (a paid service) supports distributed execution already.
+
+</div>
+
+
+## OS fine-tuning for maximum performance
+
+For the purpose of this demonstration, we are using a Linix (Ubuntu Server) machine. The instructions will be the same for any Linux distribution. 
+
+The following configuration changes are required to allow the k6 instance to use the full network capacity of the server.
+Detailed information about these settings can be found in our [OS Fine tuning article](/misc/fine-tuning-os).
+
+<div class="code-group" data-props='{}'>
+
+```bash
+sysctl -w net.ipv4.ip_local_port_range="1024 65535"
+sysctl -w net.ipv4.tcp_tw_reuse=1
+sysctl -w net.ipv4.tcp_timestamps=1
+ulimit -n 250000
+```
+
+</div>
+
+For quick testing, you can paste these commands in the root terminal window. To make these changes permanent, refer to the instructions of your Linux distribution.
+
+## Hardware considerations
+
+### Network
+Network throughput of the machine is an important consideration when running large tests. 
+Many AWS EC2 machines come with 1Gbit/s connection which may limit the amount of load k6 can generate.
+
+When running the test, you can use `iftop` in the terminal to view in real time the amount of network traffic generated.
+If the traffic is constant at 1Gbit/s, your test is probably limited by the network card. Consider upgrading to a different EC2 instance.
+
+### CPU
+Unlike many other load testing tools, k6 is heavily multithreaded. It will effectively use all available CPU cores.
+
+The amount of CPU you need depends on your test files (sometimes called test script). 
+Regardless of the test file, you can assume that large tests require significant amount of CPU power. 
+We recommend that you size the machine to have at least 20% idle cycles (up to 80% used by k6, 20% idle). 
+If k6 uses 100% to generate load, it won't have enough CPU to measure the responses correctly. 
+This may cause the result metrics to have much larger response time than in reality.
+
+### Memory
+k6 likes memory, but [it isn't as greedy as other load testing tools](https://k6.io/blog/comparing-best-open-source-load-testing-tools#memory-usage). 
+Memory consumption heavily depend on your test scenarios. To estimate the memory requirement of your test, 
+run the test on your laptop with 100VUs and multiply the consumed memory by the target number of VUs. 
```

Comment:
```suggestion
run the test on your development machine with 100VUs and multiply the consumed memory by the target number of VUs. 
```
I think "work machine" is too wide of a definition .. but not all people work/develop on laptops ;)